lib/common/socket.c

```diff
@@ -1470,3 +1470,112 @@ void h2o_sliding_counter_stop(h2o_sliding_counter_t *counter, uint64_t now)
     /* recalc average */
     counter->average = counter->prev.sum / (sizeof(counter->prev.slots) / sizeof(counter->prev.slots[0]));
 }
+
+#if H2O_USE_DTRACE && defined(__linux__)
+#include <linux/bpf.h>
+#include <linux/unistd.h>
+#include "h2o-probes.h"
+#include "include/h2o/ebpf.h"
+#include <sys/stat.h>
+
+int h2o_tracing_map_fd = 0;
+uint32_t h2o_tracing_map_last_attempt = 0;
+
+void open_map()
+{
+    // only check every second
+    uint32_t now = (uint32_t)time(NULL);
+    if (h2o_tracing_map_last_attempt == now)
+        return;
+
+    __sync_bool_compare_and_swap(&h2o_tracing_map_last_attempt, h2o_tracing_map_last_attempt, now);
+
+    // check if map exists at path
+    struct stat s;
+    if (stat(&H2O_EBPF_MAP_PATH[0], &s) == -1) {
+        // map path unavailable, cleanup fd if needed and leave
+        if (h2o_tracing_map_fd > 0) {
+            close(h2o_tracing_map_fd);
+            __sync_fetch_and_and(&h2o_tracing_map_fd, 0);
+        }
+        return;
+    }
+
+    if (h2o_tracing_map_fd > 0)
+        return; // map still exists and we have a fd
+
+    union bpf_attr attr;
+    memset(&attr, 0, sizeof(attr));
+    attr.pathname = (uint64_t)(unsigned long)&H2O_EBPF_MAP_PATH[0];
```

Comment:
I do not think we need to cast the value to `unsigned long` before casting it to `uint64_t`.

I might also argue that it's dangerous; because it would zero-clear the significant 32-bits on a LLP64 platform.